src/bosh-google-cpi/google/backendservice_service/google_backendservice_service.go

```diff
@@ -128,26 +122,39 @@ func (i GoogleBackendServiceService) RemoveInstance(vmLink string) error {
 // while trying to find the Backend Service.
 func (i GoogleBackendServiceService) find(id, region string) (BackendService, bool, error) {
 	i.logger.Debug(googleBackendServiceServiceLogTag, "Finding Google Backend Service %q", id)
-	var backendServiceItem *compute.BackendService
-	var err error
-	if region == "" {
-		backendServiceItem, err = i.computeService.BackendServices.Get(i.project, id).Do()
-	} else {
-		backendServiceItem, err = i.computeService.RegionBackendServices.Get(i.project, region, id).Do()
-	}
-	if err != nil {
-		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 {
-			return BackendService{}, false, nil
+
+	// Search for a matching backend service amongst an aggregated list containing both global and regional items
+	aggregatedBackendServices, err := i.computeService.BackendServices.AggregatedList(i.project).Do()
+	// TODO(craigatgoogle): Employ server-side name filtering once the API filter bug is fixed, https://b.corp.google.com/issues/80238913
+	if err == nil {
+		var backendService *compute.BackendService
+		for _, scopedList := range aggregatedBackendServices.Items {
+			for _, bs := range scopedList.BackendServices {
+				if bs.Name == id && (bs.Region == "" || strings.Contains(bs.Region, region)) {
```

Comment:
If we continue to increment region names (there's an eu-west4 today), `strings.Contains` would find `eu-west1` within `eu-west11`. There's a helper func somewhere around here that will extract a short region name from a fully-qualified one. Maybe worth looking into that?

Reply:
Done.